File: ios/OSVersionModule.m
//
//  OSVersionModule.m
//  AuthNative
//
//  Created by vinicius gabriel on 11/05/25.
//

#import <Foundation/Foundation.h>
#import <React/RCTBridgeModule.h>

@interface RCT_EXTERN_MODULE(OSVersionModule, NSObject)
RCT_EXTERN_METHOD(getOSVersion:(RCTPromiseResolveBlock)resolve
                  rejecter:(RCTPromiseRejectBlock)reject)
@end

File: ios/OSVersionModule.swift
//
//  OSVersionModule.swift
//  AuthNative
//
//  Created by vinicius gabriel on 11/05/25.
//

import Foundation
import UIKit

@objc(OSVersionModule)
class OSVersionModule: NSObject {


    @objc
    func getOSVersion(_ resolve: @escaping RCTPromiseResolveBlock, rejecter reject: @escaping RCTPromiseRejectBlock) {
        let version = UIDevice.current.systemVersion
        resolve(version)
    }
}
